pkg/dockerregistry/server/manifestschema1handler.go

```diff
@@ -102,7 +102,7 @@ func (h *manifestSchema1Handler) FillImageMetadata(ctx context.Context, image *i
 	}
 	image.DockerImageMetadata.Object = meta
 
-	return nil
+	return encodeRawDockerImageMetadata(image, true)
```

Comment:
above, imageMetadataFromManifest is called, which calls encodeRawDockerImageMetadata. Can't we merge the code above into imageMetadataFromManifest?